### ISSUE TITLE
`MaxLineLength` should ignore Markdown links by default

### ISSUE THREAD

**Issue by yevhenii-nadtochii** [detekt/detekt#5493]
## Expected Behavior of the rule
This issue is similar to #1070, but for Markdown links.

Consider the following example:

```kotlin
 * See: [Maven Publish Plugin | Publications](https://docs.gradle.org/current/userguide/publishing_maven.html#publishing_maven:publications)
 * https://docs.gradle.org/current/userguide/publishing_maven.html#publishing_maven:publicationspublicationspublicationspublicationspublications
```

The first line will be reported, while the second one not (thanks to #1087).

I would suggest checking if a line ends with a Markdown link as well, and don't report this case.

## Context
From [Kotlin documentation](https://kotlinlang.org/docs/kotlin-doc.html) (emphasis mine):

>> The language used to document Kotlin code (the equivalent of Java's Javadoc) is called KDoc. In its essence, KDoc combines Javadoc's syntax for block tags (extended to support Kotlin's specific constructs) and **Markdown for inline markup**.
